drizzlepac/devutils/analyze_mvm_gaia_alignment.py

```diff
@@ -102,60 +228,43 @@ def perform(mosaic_imgname, flcflt_list, diagnostic_mode=False, log_level=loguti
-    # 5a: find sources common to both the gaia table and the detection table
-    try:
-        coo_prefix_string = ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(6))
-        gaia_coo_filename = "{}_gaia.coo".format(coo_prefix_string)
-        det_coo_filename = "{}_det.coo".format(coo_prefix_string)
-        write_region_file(gaia_coo_filename, gaia_table, ['X', 'Y'], verbose=False)
-        write_region_file(det_coo_filename, detection_table, ['X', 'Y'], verbose=False)
-        matches_gaia_to_det, matches_det_to_gaia = cu.getMatchedLists([gaia_coo_filename, det_coo_filename],
-                                                                      [mosaic_imgname, mosaic_imgname],
-                                                                      [n_gaia, n_detection],
-                                                                      log_level)
-        if len(matches_gaia_to_det) == 0:
-            err_msg = "Error: No matching sources found."
-            log.error(err_msg)
-            raise Exception(err_msg)
-    finally:
-        for item in [det_coo_filename, gaia_coo_filename]:
-            if os.path.exists(item):
-                log.debug("Removing temp coord file {}".format(item))
-                os.remove(item)
-
+    # 4: compute FWHM for source finding based on sources in image
+    mpeaks, fwhm = decutils.find_point_sources(mosaic_imgname, mask=np.invert(dao_mask_array),
+                                               def_fwhm=3.0, box_size=7, block_size=(1024, 1024),
+                                               diagnostic_mode=diagnostic_mode)
+    # 5: Attempt to find matching gaia sources and userStarFinder sources first using computed FWHM value then
+    # hard-wired value.
+    fwhm_values = [fwhm, 25.0]
+    for item in enumerate(fwhm_values):
+        ctr = item[0]
+        fwhm_value = item[1]
+        detection_table, matches_gaia_to_det, matches_det_to_gaia = find_and_match_sources(fwhm_value,
+                                                                                           mosaic_hdu,
+                                                                                           mosaic_wcs,
+                                                                                           mosaic_imgname,
+                                                                                           dao_mask_array,
+                                                                                           gaia_table,
+                                                                                           xy_gaia_coords,
+                                                                                           diagnostic_mode=diagnostic_mode,
+                                                                                           log_level=log_level)
+        if len(matches_gaia_to_det) > 10:
```

Comment:
Does this limit need to be hard-coded?  Would it be useful to allow the user to set what limit they are comfortable with, or should it vary depending on how many GAIA sources actually overlap the skycell?  For example, only 8 sources overlap the HST data for p1101x11y05 out of a total of 25 for the entire skycell.

Reply:
Issue resolved by commit [094a1cd](https://github.com/spacetelescope/drizzlepac/pull/1241/commits/094a1cda714ccb414cc2f7c5d50bd96650a0866b)